src/main/java/com/opencart/pages/SearchProductResultPage.java

```diff
@@ -0,0 +1,30 @@
+package com.opencart.pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+/*
+ * Class that is useful in SearchProductTest
+ */
+
+public class SearchProductResultPage extends BasePage {
+    @FindBy(xpath = ".//div/h2")
+    private WebElement successTitle;
+
+    public WebElement getSuccessTitle() {
+        return successTitle;
+    }
+
+    public boolean foundProduct(String productName) {
+        List<WebElement> products = driver.findElements(
+                By.xpath(".//*[@class='caption']/h4"));
+        for (WebElement product : products) {
+            if (product.getText().contains(productName))
+                return true;
+        }
+        return false;
+    }
```

Comment:
I have a couple of questions about this method:
1.  Why is it located into .*Page class, why not into business logic?
2.  As I understand you want to verify if all products displayed after search contain a search phrase. But, now you make 
    a decision about successful search based only on one matching. Am I wrong?

Reply:
1. Yes, this is my fault.2.Actualy I thought that finding by this xpath.//*[@class='caption']/h4 return me all elements because it structure is the same, maybe I was wrong?
ProductContainer - It will be the class where we describe the same structure (name, description, price, tax, add to cart / wish list buttons). The poin is that we can use this class in our SearchProductPageBL and another Business logic connect with products detail?

Reply:
Currently that XPath returns you only all product names. The point is you need to create a list of ProductContainer and when you will be iterating through this list, you will check if your phrase matches the real name in the next way: 

boolean isAllProductsCorrespondToSearchPharase = true;
for (ProductContainer product : products) {
            if (!product.getName().contains(productName)) {
                isAllProductsCorrespondToSearchPharase  = false;
               }
        }
        return isAllProductsCorrespondToSearchPharase ;

Yep, the point is to use it within SearchProductPageBL and other BL classes, where it will be necessary.

Reply:
Ok, after that, we need some ProductContainerRepository that works like a Database? This will be the place where our data for each product located or not?

Reply:
In this case, we don't need it, because it is not data for our tests, like a RegisterModel, it is just WebElements on the page. And we will store it in a container, for simpler managing, validations, and interactions with that data